gremlin-go/driver/graphBinary.go

```diff
@@ -288,6 +293,200 @@ func bigIntReader(buffer *bytes.Buffer, _ *graphBinaryTypeSerializer) (interface
+		return nil, err
+	}
+	e.id = newUUID.(uuid.UUID)
+	newLabel, err := typeSerializer.read(buffer)
+	if err != nil {
+		return nil, err
+	}
+	e.label = newLabel.(string)
+	newInV, err := vertexReader(buffer, typeSerializer)
+	if err != nil {
+		return nil, err
+	}
+	e.inV = *newInV.(*Vertex)
+	newOutV, err := vertexReader(buffer, typeSerializer)
+	if err != nil {
+		return nil, err
+	}
+	e.outV = *newOutV.(*Vertex)
+	// read null bytes
+	_, _ = typeSerializer.read(buffer)
+	_, _ = typeSerializer.read(buffer)
+	return e, nil
+}
+
+//Format: {key}{value}{parent}
+func propertyWriter(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+	v := value.(*Property)
+	_, err := typeSerializer.write(v.key, buffer)
+	if err != nil {
+		return nil, err
+	}
+	_, err = typeSerializer.write(v.value, buffer)
+	if err != nil {
+		return nil, err
+	}
+	// Note that as TinkerPop currently send "references" only, parent and properties  will always be null
+	buffer.Write(nullBytes)
+	return buffer.Bytes(), nil
+}
+
+func propertyReader(buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) (interface{}, error) {
+	p := new(Property)
+	newKey, err := typeSerializer.read(buffer)
+	if err != nil {
+		return nil, err
+	}
+	p.key = newKey.(string)
+	newValue, err := typeSerializer.read(buffer)
+	if err != nil {
+		return nil, err
+	}
+	p.value = newValue
+	// read null byte
+	_, _ = typeSerializer.read(buffer)
+	return p, nil
+}
+
+//Format: {id}{label}{value}{parent}{properties}
+func vertexPropertyWriter(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+	vp := value.(*VertexProperty)
+	_, err := typeSerializer.write(vp.id, buffer)
+	if err != nil {
+		return nil, err
+	}
+	_, err = typeSerializer.write(vp.label, buffer)
+	if err != nil {
+		return nil, err
+	}
+	_, err = typeSerializer.write(vp.value, buffer)
+	if err != nil {
+		return nil, err
+	}
+	// Note that as TinkerPop currently send "references" only, parent and properties will always be null
+	buffer.Write(nullBytes)
+	buffer.Write(nullBytes)
+	return buffer.Bytes(), nil
+}
+
+func vertexPropertyReader(buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) (interface{}, error) {
+	vp := new(VertexProperty)
+	newUUID, err := typeSerializer.read(buffer)
+	if err != nil {
+		return nil, err
+	}
+	vp.id = newUUID.(uuid.UUID)
+	newLabel, err := typeSerializer.read(buffer)
+	if err != nil {
+		return nil, err
+	}
+	vp.label = newLabel.(string)
+	newValue, err := typeSerializer.read(buffer)
+	if err != nil {
+		return nil, err
+	}
+	vp.value = newValue
+	// read null byte
+	_, _ = typeSerializer.read(buffer)
+	_, _ = typeSerializer.read(buffer)
+	return vp, nil
+}
+
+//Format: {labels}{objects}
+func pathWriter(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+	p := value.(*Path)
+	_, err := typeSerializer.write(p.labels, buffer)
```

Comment:
Note that the official definition for `Path.labels` should be a list containing sets of string, so I think we'll need to have that structure here for the server to work. 
While we are leaving actual `set` implementations for later, we can add a `set` datatype here and use `listWriter`/`listReader` for it for now. This does mean that we'll need to change the `Path` struct in `graph.go` first to have `labels [][]string` instead, then propagate that change to this in the pathWriter and pathReader.